src/zeit/cms/repository/browser/menu.py

```diff
@@ -29,8 +29,15 @@ class Delete(zeit.cms.browser.menu.LightboxActionMenuItem):
 
     @property
     def visible(self):
-        return IRepositoryContent.providedBy(self.context) and \
-            not IPublishInfo(self.context).published
+        if not IRepositoryContent.providedBy(self.context):
+            return False
+        elif IPublishInfo(self.context).published:
+            return False
+        elif IFolder.providedBy(self.context):
+            for item in self.context.values():
+                if IPublishInfo(item).published:
```

Comment:
@witsch In theory: yes. The DAV server stores metadata as a triple `(path, namespace+name, value)`. So one should be able to do something like `SELECT path FROM facts WHERE url like "myfolder/*" AND name="published" AND value="yes"`.

However, as mentioned before, we can't/won't talk SQL directly, but rather go through yet another HTTP extension called the "query server", and I don't know offhand whether a) that allows the `url` in the where clause at all, and b) if it supports "startswith" or "like" or similar.

[Example queryserver usage](https://github.com/ZeitOnline/zeit.newsletter/blob/master/src/zeit/newsletter/category.py#L119), the API is unfortunately really quirky, for example you have to mention all variables you want to use in the where clause in the select clause, otherwise it silently returns no results.

Bottom line, since you've managed to cordon off that usecase (and it's a rare one), it might not be worth digging into this...